packages/kotti-ui/source/kotti-filter/components/FilterRow.vue

```diff
@@ -0,0 +1,230 @@
+<template>
+	<KtForm
+		class="kt-filter-row"
+		hideValidation
+		:isLoading="isLoading"
+		:size="Kotti.Field.Size.SMALL"
+		:value="value"
+		@input="handleSetValue"
+	>
+		<span class="kt-filter-row__prefix" v-text="prefix" />
+		<KtFieldSingleSelect
+			class="kt-filter-row__column-select"
+			formKey="key"
+			hideClear
+			:options="columnOptions"
+		/>
+		<KtFieldSingleSelect
+			class="kt-filter-row__operation-select"
+			formKey="operation"
+			hideClear
+			:isDisabled="isOperationSelectDisabled"
+			:options="operationOptions"
+		/>
+		<div :class="valueContainerClasses">
+			<component
+				:is="valueComponent"
+				:collapseTagsAfter="1"
+				formKey="value"
+				:options="valueOptions"
+				type="switch"
+			/>
+			<span v-if="showValueState" v-text="valueState" />
+		</div>
+		<ButtonLink
+			class="kt-filter-row__remove"
+			:icon="Yoco.Icon.CLOSE"
+			:isLoading="isLoading"
+			:type="Kotti.Filter.ButtonLinkType.DANGER"
+			@click="handleRemove"
+		/>
+	</KtForm>
+</template>
+
+<script lang="ts">
+import { Yoco } from '@3yourmind/yoco'
+import { computed, defineComponent } from '@vue/composition-api'
+
+import {
+	KtFieldDateRange,
+	KtFieldMultiSelect,
+	KtFieldNumber,
+	KtFieldSingleSelect,
+	KtFieldText,
+	KtFieldToggle,
+	KtForm,
+} from '../../'
+import { useTranslationNamespace } from '../../kotti-translation/hooks'
+import { Kotti } from '../../types'
+import { getOperationOptions, getValueComponent } from '../utils'
+
+import ButtonLink from './ButtonLink.vue'
+
+export default defineComponent<{
+	column: Kotti.Filter.Column
+	columnOptions: Kotti.FieldSingleSelect.Props['options']
+	isFirstItem: boolean
+	isLoading: boolean
+	value: Kotti.Filter.Filter
+}>({
+	name: 'FilterRow',
+	components: {
+		ButtonLink,
+		KtFieldDateRange,
+		KtFieldMultiSelect,
+		KtFieldNumber,
+		KtFieldSingleSelect,
+		KtFieldText,
+		KtFieldToggle,
+		KtForm,
+	},
+	props: {
+		column: {
+			default: () => ({}),
```

Comment:
What’s the use-case for an empty column? How does this match the `Kotti.Filter.Column` type?

Reply:
When adding a new filter, the component does not know which column it is until one is selected from the select control

Reply:
why not default to `null`
also ^ reiterating on Florian's comment basically, some of the keys on the `Kotti.Filter.Column` are not optional
(e.g. `key`, `label`) so you're already breaking the type?
I'd recommend `null` personally and adjusting the defintion in https://github.com/3YOURMIND/kotti/pull/388/files#diff-d8aa8b4f6cb2fe7301d9370d773dc49272579913ea04837985c576f78401232cR64 to suit it